modules/katello_composite_content_view.py

```diff
@@ -0,0 +1,163 @@
+        required: true
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    verify_ssl:
+        description:
+            - Verify SSL of the Foreman server
+        required: false
+        default: true
+        type: bool
+    name:
+        description:
+            - Name of the Katello composite content view
+        required: true
+    organization:
+        description:
+            - Organization that the Content Views are in
+        required: true
+    content_views:
+        description:
+            - List of content views that include name and version
+        required: false
+        type: list
+'''
+
+EXAMPLES = '''
+- name: "Create or update a Fedora composite content view"
+  katello_composite_content_view:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "OurFedora CCV"
+    organization: "My Cool new Organization"
+    content_views:
+      - name: 'Fedora CV'
+        version: '2.0'
+      - name: 'Ansible CV'
+        version: '1.0'
+'''
+
+RETURN = '''# '''
+
+try:
+    from ansible.module_utils.ansible_nailgun_cement import (
+        ContentView,
+        create_server,
+        find_organization,
+        find_content_views,
+        find_content_view,
+        find_content_view_version,
+        ping_server,
+        set_task_timeout,
+    )
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.foreman_helper import handle_no_nailgun
+
+
+def sanitize_composite_content_view_dict(entity_dict):
+    # This is the only true source for names (and conversions thereof)
+    name_map = {
+        'name': 'name',
+        'organization': 'organization',
+        'content_views': 'content_views',
+    }
+    result = {}
+    for key, value in name_map.items():
+        if key in entity_dict:
+            result[value] = entity_dict[key]
+    return result
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(required=True),
+            username=dict(required=True, no_log=True),
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(type='bool', default=True),
+            name=dict(required=True),
+            organization=dict(required=True),
+            content_views=dict(type='list'),
+        ),
+        supports_check_mode=True,
+    )
+
+    handle_no_nailgun(module, HAS_NAILGUN_PACKAGE)
+
+    entity_dict = dict(
+        [(k, v) for (k, v) in module.params.items() if v is not None])
+
+    server_url = entity_dict.pop('server_url')
+    username = entity_dict.pop('username')
+    password = entity_dict.pop('password')
+    verify_ssl = entity_dict.pop('verify_ssl')
+    state = entity_dict.pop('state')
```

Comment:
I am missing this in the argument_spec as well as the documentation.

Reply:
So I went and stole it (just now) from katello_product and noticed that the doc says `latest, present, absent` but the argument spec says `'present_with_defaults', 'present', 'absent'` ?